packages/web3-eth/src/index.js

```diff
@@ -485,6 +485,27 @@ var Eth = function Eth() {
                 return payload;
             }
         }),
+        // TODO: first params should be object with keys ['types', 'primaryType', 'domain', 'message']
+        // Or json.stringify version of this object
+        new Method({
+            name: 'signTypedData',
+            call: 'eth_signTypedData',
+            params: 2,
+            inputFormatter: [null, formatter.inputAddressFormatter],
+            outputFormatter: function (result) {
+                var vrs = this.accounts._decodeSignature(result);
+                return {
+                    v: vrs[0],
+                    r: vrs[1],
+                    s: vrs[2],
+                    signature: result
+                }
+            },
```

Comment:
@Amxx 

Have modified this so it resembles the `.accounts` output more. 

Could not get ganache to accept the object test input when using `JSON.stringify` as an input formatter - I've removed that. 

**Questions**
+ Could you discuss in more detail your TODO comment about the input formatting?
+ Is ganache the only client that supports this endpoint? (Geth doesn't).

Reply:
- Details about my TODO comment:

The first argument (equivalent of message) shuld be a javascript object that should contain 4 fields: 'types', 'primaryType', 'domain', 'message'. Other objects are not valid. Both 'types', 'domain' and 'message' should be objects. 'primaryType' should be a string that also happens to be a key of the 'types' field. Message should have the structure describe in types (by the 'primaryType') ... there are MANY checks that could be performed.

Of course, just JSON.stringify the object, doing the RPC call, and catching errors is possible. I'm not sure how extensive you want the pre-checks to be.

- Metamask has been the first provider to support this endpoint. Not sure about others.

I've done what I could. Feel free to ask me more or do whatever you want with my draft.

Reply:
@amxx Ok thanks!

Will add a checklist of things still needed the original PR description to make sure we cover everything here...

Reply:
Not sure how to proceed with ganache. In case that helps, here is an old piece of code I used to use with metamask:

```
function signStruct(typename, message, wallet)
{
	return new Promise((resolve, reject) => {
		web3.currentProvider.sendAsync({
			method: "eth_signTypedData_v4",
			params: [ wallet, JSON.stringify({ types: TYPES, domain: DOMAIN, primaryType: typename, message: message }) ],
			from: wallet,
		}, (err, result) => {
			if (result.error == undefined)
			{
				message.sign = result.result;
				resolve(message);
			}
			else
			{
				reject(result);
			}
		});
	});
}

```

Reply:
@amxx Thanks, that's helpful. We'll need to test vs. metamask...

Reply:
The thing with metamask is they have implemented various iterations of the EIP draft with different names. Most commonly used might be eth_signTypedData_v3, latest is eth_signTypedData_v4, not sure what eth_signTypedData fallback to (if first version that is bad)

Reply:
การเพิ่ม เเล้วข้อมูล จะอัพเดต ต้องใช้เวลา นานไหม?

Reply:
ดีครับ รอดูว่าจะ เเก้ปัญหา ตรงeth error ได้ไหม
   ผมก็มี ปัญหานี้เอง เหมือนกัน

Reply:
> ```
>   การเพิ่ม เเล้วข้อมูล จะอัพเดต ต้องใช้เวลา นานไหม?
> ```